app/utils/session_decorators.py

```diff
@@ -1,16 +1,34 @@
-from flask import session
+from flask import session, redirect, url_for
 from functools import wraps
+
 from app.models import User
 
 
+def does_user_have_session() -> bool:
+    return 'logged_in' in session and bool(User.get_by_user_id(session.get('logged_in').get('user_id')))
+
+
+def get_url_for_profile() -> str:
+    return url_for('user_routes.get_user', username=session.get('logged_in').get('username'))
+
+
 def login_required(f):
     @wraps(f)
     def decorated(*args, **kwargs):
-        if 'logged_in' in session:
-            does_user_session_exist = bool(User.get_by_user_id(session.get('logged_in')['user_id']))
-            if does_user_session_exist:
-                return f(session.get('logged_in'), *args, **kwargs)
+        if does_user_have_session():
+            return f(session.get('logged_in').get('user_id'), *args, **kwargs)
```

Comment:
This fixes another bug where we were sending the user dict instead of the user_id